dataprofiler/tests/space_time_analysis/dataset_generation.py

```diff
@@ -0,0 +1,256 @@
+    :type start_date: pd.Timestamp, optional
+    :param start_date: the latest date that datetimes can be generated at,
+        defaults to None
+    :type start_date: pd.Timestamp, optional
+
+    :return: generated datetime
+    :rtype: str
+    """
+    if not start_date:
+        # 100 years in past
+        start_date = pd.Timestamp(1920, 1, 1)
+    if not end_date:
+        # protection of 30 years in future
+        end_date = pd.Timestamp(2049, 12, 31)
+    t = rng.random()
+    ptime = start_date + t * (end_date - start_date)
+
+    return ptime.strftime(date_format)
+
+
+def random_datetimes(rng: Generator, date_format_list: str=None,
+                     start_date: str=None, end_date: str=None,
+                     num_rows: int=1) -> np.array:
+    """
+    Generate datetime given the random_state, date_format, and start/end dates.
+
+    :param rng: the np rng object used to generate random values
+    :type rng: numpy Generator
+    :param date_format: the format that the generated datatime will follow,
+        defaults to None
+    :type date_format: str, optional
+    :param start_date: the earliest date that datetimes can be generated at,
+        defaults to None
+    :type start_date: pd.Timestamp, optional
+    :param start_date: the latest date that datetimes can be generated at,
+        defaults to None
+    :type start_date: pd.Timestamp, optional
+
+    :return: array of generated datetimes
+    :rtype: numpy array
+    """
+    date_list = []
+    if not date_format_list:
+        date_format_list = ["%B %d %Y %H:%M:%S"]
+
+    for _ in range(num_rows):
+        date_format = rng.choice(date_format_list)
+        datetime = generate_datetime(rng, date_format=date_format,
+                                     start_date=start_date, end_date=end_date)
+        date_list.append(datetime)
+
+    return np.array(date_list)
+
+
+def random_categorical(rng: Generator, categories: List[str]=None,
+                       num_rows: int=1) -> np.array:
+    """
+    Randomly generates an array of categorical chosen out of categories
+
+    :param rng: the np rng object used to generate random values
+    :type rng: numpy Generator
+    :param categories: a list of values that are allowed in a categorical or None
+    :type categories: string, None
+    :param num_rows: the number of rows in np array generated
+    :type num_rows: int
+
+    :return: np array of categories
+    """
+    if categories is None:
+        categories = ["A", "B", "C", "D", "E"]
+
+    return rng.choice(categories, (num_rows,))
+
+
+def get_ordered_column(start: int=0, num_rows: int=1, **kwarg):
+    """
+    Generates an array of ordered integers
+
+    :param start: integer that the ordered list should start at
+    :type str_len_min: int
+    :param num_rows: the number of rows in np array generated
+    :type num_rows: int
+
+    :return: np array of ordered integers
+    """
+    return np.array(list(range(start, num_rows)))
```

Comment:
if we want to go straight to an array, we might be able to do:
`np.arange(start, start + num_rows)`

Reply:
will be more efficient I believe.